middle_end/flambda/naming/bindable_let_bound.ml

```diff
@@ -152,23 +145,14 @@ let add_to_name_permutation t1 ~guaranteed_fresh:t2 perm =
   | Set_of_closures { name_mode = _; closure_vars = closure_vars1; },
       Set_of_closures { name_mode = _;
         closure_vars = closure_vars2; } ->
-    let perm =
-      Closure_id.Map.fold2_stop_on_key_mismatch
-        (fun _closure_var var1 var2 perm ->
-          Name_permutation.add_fresh_variable perm
-            (Var_in_binding_pos.var var1)
-            ~guaranteed_fresh:(Var_in_binding_pos.var var2))
-        closure_vars1
-        closure_vars2
-        perm
-    in
-    begin match perm with
-    | Some perm -> perm
-    | None ->
-      Misc.fatal_errorf "Mismatching closure vars:@ %a@ and@ %a"
-        print t1
-        print t2
-    end
+    List.fold_left2
```

Comment:
We've lost the error if the numbers of closure variables are mismatched.  Maybe we could use `List.compare_lengths` to resurrect it.

Reply:
Good idea; done.